Automattic/wp-calypso
# Media: Clear icon site value and setting upon deleting associated media

Closes #9939

This pull request seeks to clear site icon values (both on the site object itself and the site setting) when deleting a media item which is currently assigned as the site icon. Server-side, the option is automatically cleared when the media is deleted, so we only need to recreate this behavior in the client ([related source](https://github.com/WordPress/WordPress/blob/d94530e721f7bf7ff04096a889046aaa33836f73/wp-admin/includes/class-wp-site-icon.php#L202-L216)).

__Testing Instructions:__

Testing will be made easier by use of [Redux DevTools Chrome extension](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en).

1. Navigate to [Site Settings](http://calypso.localhost:3000/settings/general)
2. Select a site
3. Assign a site icon by clicking "Change Site Icon" (select image, "Continue", crop, "Done")
  - This will set the site setting value, but does not yet change front-end icon shown
4. Click "Change Site Icon" again
5. Choose the newly created cropped image
6. Click Delete
7. In developer tools Redux tab, note that...
  - Upon the `MEDIA_DELETE` dispatched action, the diff shows that the `state.siteSettings.items[  siteId ].site_icon` was set to null
  - A `RESET_SITE_ICON` action is also dispatched, and the diff shows that the `state.sites.items[ siteId ].icon` was removed (but only if you have a site icon already assigned via non-Calypso flow)

You can also verify that deleting any other media will dispatch `MEDIA_DELETE` with no effect, and will not dispatch `RESET_SITE_ICON`.
